tests/test_training_pipeline.py

```
@@ -0,0 +1,146 @@
+    )
+
+    assert engine.calls == []
+
+
+@pytest.mark.asyncio
+async def test_training_workflow_exits_immediately_on_non_positive_max_cycles() -> None:
+    settings = get_settings().model_copy(update={"training_cycle_jitter_seconds": 0})
+    for limit in (0, -1):
+        engine = _StubEngine()
+        await training_workflow(
+            engine_factory=lambda: engine,
```

Ruff (0.13.2)

Function definition does not bind loop variable `engine`

(B023)